submissions/krylenger/frogger/app.js

```diff
@@ -0,0 +1,246 @@
+  if (this.y === borders.topY) {
+    setTimeout(() => {
+        this.score += 2;
+        score.textContent = `Score: ${this.score}`
+      alert(`Well done! Your score is ${this.score}. Keep it up!`);
+      this.y = borders.bottomY;
+    }, 13);
+  }
+};
+
+Player.prototype.handleInput = function(keyPress) {
+    switch (keyPress) {
+      case "up":
+        if (this.y > borders.topY) {
+          this.y -= tileDimensions.height;
+        }
+        break;
+      case "down":
+        if (this.y < borders.bottomY) {
+          this.y += tileDimensions.height;
+        }
+        break;
+      case "left":
+        if (this.x > borders.leftX) {
+            this.x -= tileDimensions.width;
+        }
+        break;
+      case "right":
+        if (this.x < borders.rightX) {
+            this.x += tileDimensions.width;
+        }
+        break;
+    }
+}
+
+Player.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    
+}
+
+let player = new Player(220, 464);
+
+//////////  ITEM /////////////////
+
+const randomPositionInArray = (arrLength) => {
+    return (Math.floor(Math.random() * arrLength));
+}
+
+const randomPosition = () => {
+  const x = [0, 100, 200, 300, 400];
+  const y = [132, 215, 298];
+
+  const randomX = x[randomPositionInArray(x.length)];
+  const randomY = y[randomPositionInArray(y.length)];
+  return { x: randomX, y: randomY };
+};
+
+const Item = function() {
+    this.rndPosition = randomPosition();
+    this.x = this.rndPosition.x;
+    this.y = this.rndPosition.y;
+    this.width = 100;
+    this.height = 83;
+}
+
+Item.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+}
+
+//////////  STAR /////////////////
+
+const Star = function() {
+    Item.apply(this, arguments);
+    this.sprite = 'images/Star.png';
+}
+
+Star.prototype = Object.create(Item.prototype);
+Star.prototype.constructor = Star;
+
+Star.prototype.update = function() {
+    if (this.y === player.y && this.x === player.x - 20) {
+        this.x = randomPosition().x;
+        this.y = randomPosition().y;
+        setTimeout(() => {
+            player.score += 1;
+            score.textContent = `Score: ${player.score}`;
+        }, 100);
+    }
+}
+
+//////////  HEART /////////////////
+
+const Heart = function() {
+    Item.apply(this, arguments);
+    this.sprite = 'images/Heart2.png';
+}
+
+Heart.prototype = Object.create(Item.prototype);
+Heart.prototype.constructor = Heart;
+
+Heart.prototype.update = function() {
+    if (this.y === player.y && this.x === player.x - 20) {
+        this.x = randomPosition().x;
+        this.y = randomPosition().y;
+        setTimeout(() => {
+            console.log('hi');
+            player.lives += 1;
+            lives.textContent = `Lives: ${player.lives}`
+        }, 13);
+    }
+}
+
+let star = new Star();
+let heart = new Heart();
+
+
+document.addEventListener('keyup', function(e) {
+    var allowedKeys = {
```

Comment:
Let's avoid mixing `var` with `let/const` with not special reason

Reply:
✅  Done